modules/private.php

```diff
@@ -0,0 +1,298 @@
+			$output .= "Disallow: /\n";
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Filters the link title attribute for the message displayed in the 'At a Glance' dashboard widget.
+	 *
+	 * @param string $text Default attribute text.
+	 */
+	static function privatize_privacy_on_link_title( $text ) {
+		if ( Jetpack_Private::is_private_blog() ) {
+			return __( 'Your site is visible only to registered members', 'jetpack' );
+		}
+
+		return $text;
+	}
+
+	/**
+	 * Filters the link label for the message displayed in the 'At a Glance' dashboard widget.
+	 *
+	 * @param string $text Default text.
+	 */
+	static function privatize_privacy_on_link_text( $text ) {
+		if ( Jetpack_Private::is_private_blog() ) {
+			return __( 'Private', 'jetpack' );
+		}
+
+		return $text;
+	}
+
+	/**
+	 * Output the meta tag that tells Pinterest not to allow users to pin
+	 * content from this page.
+	 * https://support.pinterest.com/entries/21063792-what-if-i-don-t-want-images-from-my-site-to-be-pinned
+	 */
+	static function private_no_pinning() {
+		echo '<meta name="pinterest" content="nopin" />';
+	}
+
+	/**
+	 * Prevents ajax requests on private blogs for users who don't have permissions
+	 *
+	 * @param string    $action The Ajax nonce action.
+	 * @param false|int $result The result of the nonce check.
+	 */
+	static function private_blog_ajax_nonce_check( $action, $result ) {
+		global $current_user, $wpdb;
+
+		if ( is_super_admin() || ! Jetpack_Private::is_private_blog() ) {
+			return;
+		}
+
+		if ( 1 !== $result && 2 !== $result ) {
+			return;
+		}
+
+		if ( 'find-posts' !== $action && 'internal-linking' !== $action ) {
+			return;
+		}
+
+		// Make sure we are in the right code path, if not bail now.
+		if ( ! is_admin() || ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {
+			return;
+		}
+
+		if ( ! Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) {
+			wp_die( -1 );
+		}
+	}
+
+	/**
+	 * Disables WordPress Rest API for external requests
+	 */
+	static function disable_rest_api() {
+		global $pagenow, $current_user, $wpdb;
+
+		if ( ! Jetpack_Private::is_private_blog() ) {
+			return;
+		}
+
+		if ( $current_user && ( is_super_admin() || Jetpack_Private::is_private_blog_user( $wpdb->blogid, $current_user ) ) ) {
+			return;
+		}
+
+		die( __( 'This site is private.', 'jetpack' ) );
+	}
+
+	static function module_override( $modules ) {
+		$disabled_modules = array(
+			'publicize',
+			'sharedaddy',
+			'subscriptions',
+		);
+
+		foreach ( $disabled_modules as $module_slug ) {
+			$found = array_search( $module_slug, $modules );
+			if ( false !== $found ) {
+				unset( $modules[ $found ] );
+			}
+		}
+
+		return $modules;
+	}
+
+	/**
+	 * Prevent changes to the blog_public option when the module is enabled
+	 */
+	static function prevent_update_option_blog_public() {
+		add_settings_error( 'general', 'setting_not_updated', __( "Can't update this setting —	 it's being mananged by Jetpack.", 'jetpack' ), 'error' );
```

Comment:
I got that warning to show when I attempted to turn on the private option in Settings > Reading:

<img width="568" alt="screenshot 2018-11-27 at 17 30 55" src="https://user-images.githubusercontent.com/426388/49096198-5f0ddf80-f26a-11e8-8ba9-58c853e40e71.png">

Yet when I refreshed the page, the settings were in fact saved. But now I can't seem to be able to actually revert to a public status. I get the error when I try to change the settings, but no settings are saved.

Reply:
I changed the way this works. Now we don't rely on the `blog_public` setting, so you can change the setting, but it will be ignored. I added some messages to make that clear.